plugins/service.js

```diff
@@ -0,0 +1,59 @@
+const Plugin = require('./base');
+const { Scopes } = require('../lib/constants');
+
+const setContext = (properties, service) => {
+    const merge = (acc, [key, to]) =>
+        !properties[to] && service[key] ?
+            Object.assign(acc, { [to]: service[key] }) : acc;
+    return Object.entries({
+        'name': 'service.name',
+        'version': 'service.version',
+        'instanceId': 'service.instance.id',
+        'namespace': 'service.namespace'
+    }).reduce(merge, properties);
+};
+
+const associateContext = (resource, service) => (connect) =>
+    (url, { clientProperties = {}, ...socketOptions } = {}) => {
+        if (!resource) {
+            clientProperties = setContext(clientProperties, service);
+        }
+        return connect(url, {
+            clientProperties,
+            ...socketOptions
+        });
+    };
+
+const queueName = (binding, exchange, service) => {
+    let name = `${service.name}:`;
+
+    if (typeof binding === 'string') name += binding;
+    else name += JSON.stringify(binding || null);
+
+    return name += exchange ? `:${exchange}` : '';
```

Comment:
Why the exchange is used in the queue name? I was thinking each exchange was kind of isolated so we would not need to use it to differentiate queues

Reply:
But it doesn't mean routing keys never coincide (like if one use exchanges for namespacing it would the case)

Reply:
Indeed but as user would need to specify the exchange when publishing, it should not cause any trouble

Just trying to keep it as minimal as possible, what would be the drawback to not have it?

Reply:
I'm less certain now, but would
```node
client.type('direct').subscribe('foo', (msg) => {});
client.type('topic').subscribe('foo', (msg) => {});
```
not create bindings between a queue `service.a:foo` and two exchanges? Or is it what we expect?

Reply:
I did some testing and you are right, I got a bad idea of the relationship between a queue and an exchange :+1: 

I made a few testing on my side and confirm it all works well

Reply:
Thank you for checking!

I would still like to keep an eye on this, so any important case is overlooked.